pkg/controller/migmigration/hooks.go

```diff
@@ -0,0 +1,212 @@
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (t *Task) runHooks() (int32, error) {
+	var hookRef *corev1.ObjectReference
+	var client k8sclient.Client
+	var err error
+
+	switch t.Phase {
+	case PreBackupHooks:
+		hookRef = t.PlanResources.MigPlan.Spec.Hooks.PreBackup
+	case PostBackupHooks:
+		hookRef = t.PlanResources.MigPlan.Spec.Hooks.PostBackup
+	case PreRestoreHooks:
+		hookRef = t.PlanResources.MigPlan.Spec.Hooks.PreRestore
+	case PostRestoreHooks:
+		hookRef = t.PlanResources.MigPlan.Spec.Hooks.PostRestore
+	default:
+		err := fmt.Errorf("Unknown Hook Phase %s", t.Phase)
+		log.Trace(err)
+		return -1, err
+	}
+
+	hook := migapi.MigHook{}
+	job := &batchv1.Job{}
+
+	if hookRef != nil {
+		err = t.Client.Get(
+			context.TODO(),
+			types.NamespacedName{
+				Name:      hookRef.Name,
+				Namespace: hookRef.Namespace,
+			},
+			&hook)
+		if err != nil {
+			log.Trace(err)
+			return -1, err
+		}
+
+		switch hook.Spec.TargetCluster {
+		case "destination":
+			client, err = t.getDestinationClient()
+			if err != nil {
+				log.Trace(err)
+				return -1, err
+			}
+		case "source":
+			client, err = t.getSourceClient()
+			if err != nil {
+				log.Trace(err)
+				return -1, err
+			}
+		default:
+			err := fmt.Errorf("targetCluster myst be 'source' or 'destination'. %s unknown", hook.Spec.TargetCluster)
+			log.Trace(err)
+			return -1, err
+		}
+
+		if hook.Spec.Playbook == "" {
+			job = t.baseJobTemplate(hook)
+		} else {
+			playbookData, err := base64.StdEncoding.DecodeString(hook.Spec.Playbook)
+			if err != nil {
+				log.Trace(err)
+				return -1, err
+			}
+
+			configMap := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: hook.Spec.ExecutionNamespace,
+					Name:      strings.ToLower(t.PlanResources.MigPlan.Name + "-" + t.Phase),
+				},
+				BinaryData: map[string][]byte{
+					"playbook.yml": []byte(playbookData),
+				},
+			}
+
+			err = client.Get(context.TODO(),
+				types.NamespacedName{
+					Name:      configMap.ObjectMeta.Name,
+					Namespace: configMap.ObjectMeta.Namespace,
+				},
+				configMap)
+			if err != nil && apierrors.IsNotFound(err) {
+				err = client.Create(context.TODO(), configMap)
+				if err != nil {
+					log.Trace(err)
+					return -1, err
+				}
+			} else if err != nil {
+				log.Trace(err)
+				return -1, err
+			}
+			job = t.playbookJobTemplate(hook)
+		}
+
+		err = client.Get(context.TODO(),
+			types.NamespacedName{
+				Name:      job.ObjectMeta.Name,
+				Namespace: job.ObjectMeta.Namespace,
+			},
+			job)
+		if err != nil && apierrors.IsNotFound(err) {
+
+			err = client.Create(context.TODO(), job)
+			if err != nil {
+				log.Trace(err)
+				return -1, err
+			}
+		} else if err != nil {
+			log.Trace(err)
+			return -1, err
+		} else if job.Status.Succeeded == 1 {
+			return job.Status.Succeeded, nil
+		}
+	}
+
+	return 1, nil
```

Comment:
It will. I've already addressed this. I don't know why it was working at all before changes I made today. ![image](https://user-images.githubusercontent.com/1850500/77448906-ba50d300-6dc7-11ea-9703-c1921fb018bb.png)